unixfs/io/dagreader.go

```diff
@@ -77,3 +85,332 @@ func NewDagReader(ctx context.Context, n ipld.Node, serv ipld.NodeGetter) (DagRe
 		return nil, ErrUnkownNodeType
 	}
 }
+
+// NewDagReaderWithSize constructs a new `dagReader` with the file `size`
+// specified.
+func NewDagReaderWithSize(ctx context.Context, n ipld.Node, serv ipld.NodeGetter, size uint64) (DagReader, error) {
```

Comment:
It was a little unclear to me at first that this was the recalculated size of the whole file/DAG — I thought it might be the maximum amount of data to read.

Reply:
Yes, this was an ugly patch I had to introduce to preserve the `Size()` API, I'll try to document it more clearly.